### DISCUSSION TITLE
mvc-pager-update/

### DISCUSSION THREAD

**Discussion by martijnboland** [martijnboland/blog#40] [Blog comments]
# mvc-pager-update/

https://blogs.taiga.nl/martijn/posts/mvc-pager-update/

**Comment by disqus-comments-migrator** [comment]
<em>sebastian commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at februari 13 2011, 10:47</em>

---

<p>Dear Martijn, <br>thank you so much for your mvcpager. It was very important for my website development. I would like to ask you what I have to do if instead of categoryName using string I need to call categoryId from db.categories?. I cannot move forward with that receiving error "this dictionary requires a model item of type IPagedList`1[Social.ViewModels.ProductViewModelList]"<br>thank you so much for your time.</p>

**Comment by disqus-comments-migrator** [comment]
<em>martijn commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at februari 14 2011, 03:02</em>

---

<p>@sebastian: in your situation I'd create a view model for the page that has a SelectList of categories in it and a also an IPagedList with the results. The categories are displayed as</p><p>&lt;%= Html.DropDownList("CategoryId", Model.Categories) %&gt;</p><p>and the pager would be something like</p><p>&lt;%= Html.Pager(Model.Results.PageSize, Model.Results.PageNumber, Model.Results.TotalItemCount, new { CategoryId = Model.Categories.SelectedValue } )%&gt;</p>

**Comment by disqus-comments-migrator** [comment]
<em>sebastian commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at februari 14 2011, 09:06</em>

---

<p>Hi Martijn, thank you so much for your prompt response. I created the following controller taking into consideration your example</p><p>public ActionResult Browse(int categoryId, int? page)<br>        {<br>            const int defaultPageSize = 20;<br>            IList allProducts = new List();<br>            categoryName = ?</p><p>            int currentPageIndex = page.HasValue ? page.Value - 1 : 0;<br>            var products = allProducts.Where(p =&gt; p.CategoryId.Equals(categoryId)).ToPagedList(currentPageIndex, defaultPageSize);</p><p>            ViewData["CategoryDisplayName"] = categoryId;<br>            return View(products);<br>}</p><p>but on categorydisplayname you took the value from categoryname which is a string or ?? this.categories . In the case that you need to change that value for a category model, retrieving and integer what do you suggest to do?. sorry if its a newbie question but I´m learning with your example.<br>thank you so much,brgds</p>

**Comment by disqus-comments-migrator** [comment]
<em>sebastian commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at februari 20 2011, 11:31</em>

---

<p>Hi Martijn, I´ve been trying but I cannot move forward in my implementation of the pager. I´m only trying to display a list of products (from db.product) filtered by categoryid (from db.category). if categoryid=2 display those products paginated. I tried to use the example with string details but with no success. I dont know what i have to do in order to use the mvc pager sucessfully. Thanks in advance for any help.<br>brgds.</p>

**Comment by disqus-comments-migrator** [comment]
<em>martijn commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at februari 21 2011, 03:35</em>

---

<p>@sebastian: could you please try the solution that I proposed earlier?</p><p>You need to create a class (ViewModel) that contains both the list of categories (as SelectList) and the results as IPagedList. <br>You might also add the selected categoryId as property of the view model if the SelectedValue doesn't work. Hopefully this will help you a little.</p>

**Comment by disqus-comments-migrator** [comment]
<em>sebastian commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at februari 21 2011, 03:42</em>

---

<p>ok, I will do that. thank you!</p>
